backend/app/routers/files.py

```diff
@@ -155,17 +150,46 @@ async def add_file_entry(
     )
 
 
+async def add_local_file_entry(
+    new_file: FileDB,
+    user: UserOut,
+    es: Elasticsearch,
+    rabbitmq_client: BlockingChannel,
+    content_type: Optional[str] = None,
+):
+    """Insert FileDB object into MongoDB (makes Clowder ID), then Minio (makes version ID), then update MongoDB with
+    the version ID from Minio.
+
+    Arguments:
+        file_db: FileDB object controlling dataset and folder destination
+        file: bytes to upload
+    """
+
+    content_type_obj = get_content_type(new_file.name, content_type)
+    new_file.content_type = content_type_obj
+    await new_file.insert()
+
+    # Add entry to the file index
+    await index_file(es, FileOut(**new_file.dict()))
+
+    # TODO - timing issue here, check_feed_listeners needs to happen asynchronously.
+    time.sleep(1)
```

Comment:
just curious, isn't all those await already assume it's synchronous event, why do we need to still add 1 second of waiting time?

Reply:
I also get the file version error.
When files are uploaded a FileVersion object is added to the mongodb, but this does not happen for local files. Not sure what the best fix would be.

Reply:
@longshuicy It seems there is a slight delay between indexing a file in ES and having ES properly return the file in a search. even though we await the insert, I was still having my search return 0 results if I didn't wait a second to perform the search. It's possible we could reduce this time or (better option) move the automatic triggering logic to some asynchronous process in the background unconnected to this endpoint.

Reply:
Added a check to avoid looking up version 0 of metadata, for local etc. files that have version = 0 this is treated the same as None (get latest metadata instead). Should fix this for both getting/listing and posting/editing user metadata to lcoal files.